go/iceberg/schema_test.go

```diff
@@ -0,0 +1,795 @@
+				ValueType: &iceberg.StructType{
+					Fields: []iceberg.NestedField{
+						{ID: 11, Name: "age", Type: iceberg.PrimitiveTypes.Int32, Required: true},
+					},
+				},
+				ValueRequired: true,
+			},
+		})
+
+	assert.Truef(t, sc.Equals(expected), "expected: %s\ngot: %s", expected, sc)
+}
+
+func TestPruneColumnsSelectOriginalSchema(t *testing.T) {
+	id := tableSchemaNested.HighestFieldID()
+	selected := make(map[int]iceberg.Void)
+	for i := 0; i < id; i++ {
+		selected[i] = iceberg.Void{}
+	}
+
+	sc, err := iceberg.PruneColumns(tableSchemaNested, selected, true)
+	require.NoError(t, err)
+
+	assert.True(t, sc.Equals(tableSchemaNested))
+}
+
+func TestPruneNilSchema(t *testing.T) {
+	_, err := iceberg.PruneColumns(nil, nil, true)
+	assert.ErrorIs(t, err, iceberg.ErrInvalidArgument)
+}
+
+func TestSchemaRoundTrip(t *testing.T) {
+	data, err := json.Marshal(tableSchemaNested)
+	require.NoError(t, err)
+
+	assert.JSONEq(t, `{
+		"type": "struct",
+		"schema-id": 1,
+		"identifier-field-ids": [1],
+		"fields": [
+			{
+				"type": "string",
+				"id": 1,
+				"name": "foo",
+				"required": false
+			},
+			{
+				"type": "int",
+				"id": 2,
+				"name": "bar",
+				"required": true
+			},
+			{
+				"type": "boolean",
+				"id": 3,
+				"name": "baz",
+				"required": false
+			},
+			{
+				"id": 4,
+				"name": "qux",
+				"required": true,
+				"type": {
+					"type": "list",
+					"element-id": 5,
+					"element-required": true,
+					"element": "string"
+				}
+			},
+			{
+				"id": 6,
+				"name": "quux",
+				"required": true,
+				"type": {
+					"type": "map",
+					"key-id": 7,
+					"key": "string",
+					"value-id": 8,
+					"value": {
+						"type": "map",
+						"key-id": 9,
+						"key": "string",
+						"value-id": 10,
+						"value": "int",
+						"value-required": true
+					},
+					"value-required": true
+				}
+			},
+			{
+				"id": 11,
+				"name": "location",
+				"required": true,
+				"type": {
+					"type": "list",
+					"element-id": 12,
+					"element-required": true,
+					"element": {
+						"type": "struct",
+						"fields": [
+							{
+								"id": 13,
+								"name": "latitude",
+								"type": "float",
+								"required": false
+							},
+							{
+								"id": 14,
+								"name": "longitude",
+								"type": "float",
+								"required": false
+							}
+						]
+					}
+				}
+			},
+			{
+				"id": 15,
+				"name": "person",
+				"required": false,
+				"type": {
+					"type": "struct",
+					"fields": [
+						{
+							"id": 16,
+							"name": "name",
+							"type": "string",
+							"required": false
+						},
+						{
+							"id": 17,
+							"name": "age",
+							"type": "int",
+							"required": true
+						}
+					]
+				}
+			}
+		]
+	}`, string(data))
+
+	var sc iceberg.Schema
+	require.NoError(t, json.Unmarshal(data, &sc))
+
+	assert.Truef(t, tableSchemaNested.Equals(&sc), "expected: %s\ngot: %s", tableSchemaNested, &sc)
+}
+
+func TestRemainingTypes(t *testing.T) {
```

Comment:
I think there should be a `types_test.go` where this should live and where we can test all types with the JSON parsing, similar to how it's done on the Python side in `test_types.py`